templates/ovncontroller/bin/stop-vswitchd.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# Copyright 2024 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+set -ex
+source $(dirname $0)/functions
+
+# Clean up any previously created flow backups to avoid conflict with newly
+# generated backup.
+cleanup_flows_backup
+
+# Passing --real to mimic what upstream startup scripts do; maybe redundant.
+bridges=$(ovs-vsctl -- --real list-br)
+
+# Saving flows to avoid disrupting gateway datapath.
+mkdir $FLOWS_RESTORE_DIR
+TMPDIR=$FLOWS_RESTORE_DIR /usr/share/openvswitch/scripts/ovs-save save-flows $bridges > $FLOWS_RESTORE_SCRIPT
```

Comment:
Just wondering if save-flows doesn't take much time if there are very large number of flows, considering it do not bypass default terminate timeout of 30 seconds as we also hold ovsdb-server prestop script

Reply:
That's an interesting question! I will check with OVS folks, in the meantime I think we can stick to the default 30 seconds. We can bump them if we learn we need more time sometimes. (`terminationGracePeriodSeconds` pod config.)

save-flows is part of systemd unit restart procedure, so we can also get some guidance for applicable timeouts from what systemd does.

Reply:
AFAIU upstream uses 300 seconds for systemd units: https://github.com/openvswitch/ovs/blob/6b09799f03548f0377788f18160a3d9db3eac752/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in#L32

Reply:
Ilya M from OVS team says that under normal conditions, it should take much less time than 30 seconds. I guess we are fine.